lib/utils.nix

```diff
@@ -125,27 +125,93 @@ rec {
     end
   '';
 
+  /**
+    Convert the given String to a Lua [long literal].
+
+    [long literal]: https://www.lua.org/manual/5.4/manual.html#3.1
+
+    # Examples
+
+    ```nix
+    nix-repl> toLuaLongLiteral "simple"
+    "[[simple]]"
+    ```
+
+    ```nix
+    nix-repl> toLuaLongLiteral "]]"
+    "[=[]]]=]"
+    ```
+
+    # Type
+
+    ```
+    toLuaLongLiteral :: String -> String
+    ```
+  */
+  toLuaLongLiteral =
+    string:
+    let
+      buildInfix =
+        depth:
+        let
+          infix = lib.strings.replicate depth "=";
+        in
+        if lib.hasInfix "]${infix}]" string then buildInfix (depth + 1) else infix;
+
+      infix = buildInfix 0;
+    in
+    "[${infix}[${string}]${infix}]";
```

Comment:
@MattSturgeon, .ooking at this one more time: one unfortunate fact of this implementation is that we've repeated the "how to build a close token" logic in 2 places, and it's really important that they stay in sink.

I'm going to rework this a little bit. Standby...

Reply:
It is a little ugly, but I don't think it is a major issue.

If you don't like it, you could use the previous approach of returning a `{open,close}` attrset. Or you could do something like `builtins.replaceStrings ["]"] ["["] closingToken`, which is also ugly, but less redundant.

Reply:
Updated. @MattSturgeon, please take another look.